src/core/algorithms/md/hymd/preprocessing/similarity_measure/monge_elkan_metric.cpp

```diff
@@ -0,0 +1,22 @@
+#include "algorithms/md/hymd/preprocessing/similarity_measure/monge_elkan_metric.h"
+
+#include <algorithm>
+#include <functional>
+#include <limits>
+
+double MongeElkan(
+        std::vector<std::string> const& a, std::vector<std::string> const& b,
+        std::function<double(std::string const&, std::string const&)> similarityFunction) {
```

Comment:
```suggestion
        auto similarityFunction) {
```
Make it accept a template like this and pull to header file